### PR TITLE
add support of i18n

### PR DESCRIPTION
**Pull request by Kaijun**
As in https://github.com/crate/crate-admin/pull/203 described, i just open this PR against master for adding i18n support.

In addition, what @chaudum mentioned in https://github.com/crate/crate-admin/pull/203 are also fixed

https://github.com/crate/crate-admin/commit/87e020c4ad423db29395950d6de30d34e206117a fixed: Naming https://github.com/crate/crate-admin/pull/203#issuecomment-226487287 cursor style https://github.com/crate/crate-admin/pull/203#issuecomment-226769407 

https://github.com/crate/crate-admin/commit/4fd49d9fc5dd2dcf5e491f63aef8bb66267a1d12 added persistence of chosen language: https://github.com/crate/crate-admin/pull/203#issuecomment-226773770

Notice: need to `bower install` before starting server.

### CODE REVIEW

File: app/scripts/controllers/console.js [hunk 1]
```diff
@@ -62,14 +62,17 @@ angular.module('console', ['sql'])
...
-          var history = JSON.parse(localStorage.getItem("crate.console.query_list") || '[]');
-          localStorage.setItem("crate.console.query_list", JSON.stringify([]));
-          inputScope.recentCursor = 0;
-          recentQueries = [];
-          var msg = history.length == 1 ? "1 entry in console history has been cleared." :
-            history.length + " entries in console history have been cleared.";
-          $scope.info.message = msg;
-          $scope.info.hide = false;
+          $translate(['CONSOLE.CLEAR_HISTORY_MSG', 'CONSOLE.CLEAR_HISTORY_MSG_PLURAL']).then(function(i18n) {
+            console.log(i18n)
```
Comment (chaudum) [inline app/scripts/controllers/console.js]: please remove `console.log`

File: app/scripts/controllers/common.js [hunk 2]
```diff
@@ -87,6 +87,23 @@ var commons = angular.module('common', ['stats', 'udc'])
...
+
+      if (urlPattern.charAt(0) != '/') {
+        urlPattern = "/" + urlPattern;
+      }
+
+      var queryElements = this.navBarElements.filter(function(item) {
+        return item.urlPattern==urlPattern;
+      });
+
+      if(queryElements.length===0){
```
Comment (chaudum) [inline app/scripts/controllers/common.js]: code formatting:
```js
if (queryElements.length === 0) {
    return;
}
```

File: app/scripts/controllers/common.js [hunk 3]
```diff
@@ -157,12 +174,32 @@ var commons = angular.module('common', ['stats', 'udc'])
       });
     }
   });
+})
+.controller('LanguageSwitchController', function ($scope, $translate) {
+  $scope.changeLanguage = function (langKey) {
+    // Check if AutoDetect is chosen, if yes, set langKey as preferredLanguage.
+    langKey = langKey==='auto' ? $translate.preferredLanguage() : langKey;
```
Comment (chaudum) [inline app/scripts/controllers/common.js]: code formatting: see example above

File: app/views/statusbar.html [hunk 4]
```diff
@@ -1,37 +1,47 @@
 <div class="statusbar" ng-controller="StatusBarController">
   <ul class="nav navbar-nav navbar-right navbar-user">
 
-    <li>Cluster: <span class="label label-cluster">{{ cluster_name }}</span></li>
-    <li>Version: <span ng-class="{'label-danger':version_warning}" ng-show="version" data-original-title="{{ versions.join('\n') }}" rel="tooltip">{{ version.number }}<span ng-show="version.snapshot">-SNAPSHOT-{{ version.hash.substr(0,7) }}</span></span></li>
-    <li>Nodes: <span class="label">{{ num_nodes }}</span></li>
+    <li>{{ 'STATUSBAR.CLUSTER' | translate }}: <span class="label label-cluster">{{ cluster_name }}</span></li>
+    <li>{{ 'STATUSBAR.VERSION' | translate }}: <span ng-class="{'label-danger':version_warning}" ng-show="version" data-original-title="{{ versions.join('\n') }}" rel="tooltip">{{ version.number }}<span ng-show="version.snapshot">-{{ 'STATUSBAR.SNAPSHOT' | translate }}-{{ version.hash.substr(0,7) }}</span></span></li>
+    <li>{{ 'STATUSBAR.NODES' | translate }}: <span class="label">{{ num_nodes }}</span></li>
     <li>Health:
```
Comment (chaudum) [inline app/views/statusbar.html]: translation of Health missing

**Comment by chaudum** [comment]
@Kaijun looks really great!

I've left a few minor comments. When can we expect the Chinese translation? Or should we remove it for this PR and make a separate PR for it?
Here is the German translation you can use in this PR: https://gist.github.com/chaudum/b3dbac9e9d213d3df5472da7192d1a40
Oh, one last thing I almost forgot: can you also add an entry in the [changelog](https://github.com/crate/crate-admin/blob/master/CHANGES.txt#L7) (underneath unreleased section).

Thanks a lot for your contribution!

**Comment by Kaijun** [comment]
@chaudum  Could you please help me to finish the rest of fixes and merge your German translation in? 
I would rather open a  new separate PR for the Chinese translation.

**Comment by chaudum** [comment]
@Kaijun ok will do that!

**Comment by chaudum** [comment]
@Kaijun your changes were merged into master with #206 
once again, thanks for your contribution and we're looking forward to your chinese translation :)

**Pull request closed without merging**